.circleci/config.yml

```diff
@@ -1,19 +1,32 @@
 version: 2
 jobs:
   build:
-    docker:
-      - image: circleci/python:2.7-stretch
+    machine:
+      image: circleci/classic:latest
+    environment:
+      GOOGLE_PROJECT_ID: ci-test-data-explorer
     steps:
       - checkout
       - restore_cache:
           key: virtualenv-{{ .Branch }}-{{ checksum "bigquery/requirements.txt" }}
       - run:
           name: Run the yapf python linter
           command: |
-            virtualenv ~/yapf
-            source ~/yapf/bin/activate
             pip install yapf
             yapf -dr .
+      - run:
+          name: Run the BigQuery indexer
+          command: |
+            echo ${GOOGLE_SERVICE_KEY} | base64 --decode > ${HOME}/.config/gcloud/application_default_credentials.json
+            cd bigquery
+            docker network create dataexplorer_default
+            docker-compose up -d elasticsearch
```

Comment:
Just have a single `docker-compose up` command. Replace 23-24 with:
`BILLING_PROJECT_ID=${GOOGLE_PROJECT_ID} docker-compose up --build'

Reply:
`--exit-code-from indexer` captures the exit code from the indexer such that CircleCI passes or fails this step dependent on that (otherwise even if the indexer fails CircleCI says success). The effect of this is that `docker-compose up --build indexer` runs to completion and then exits.

If we try to bundle that together with `docker-compose up -d elasticsearch`, then docker brings up both images, runs the indexer, and exits, making further examination/testing of the Elasticsearch instance impossible. Instead, I `docker-compose up -d elasticsearch` to spin up Elasticsearch (equivalent of a k8s service) and then separately `docker-compose up --exit-code-from indexer --build indexer` (equivalent of a k8s job) such that CircleCI fails if the indexer fails but keeps Elasticsearch running so we can make requests to it for testing purposes.

Reply:
Discussed offline. This is the simplest for me to understand and maintain:
```
- BigQuery section
  - docker-compose up --build
  - test
  - docker-compose stop
- GCS section
  - docker-compose up --build
  - test
  - docker-compose stop
```
Pulling ES image is slow, but starting ES should be less than 10 minutes. So starting ES for GCS should be fast, since image is already on vm.

If something fails:
- In CircleCI, click on Rebuild with SSH
- ssh in VM
- Restart ES container. It will have the index from the indexer run.

Reply:
Maybe don't need `docker-compose stop`